### ISSUE TITLE
Expect for references + "attribute" spec method

### ISSUE THREAD

**Issue by xinitrc86** [EduardoCopat/aspec#20]
Hi Copat, 

I've found myself doing a helper class with expects and remembered this project. 
It does not fit my need, so maybe we could add this as an idea. 

**Problem**
Working with references to data require some boilerplate code to be able to check values. 

**Proposed solution**
Additional spec method 'attribute'.
An expectant that receives a reference to data.

```
expect( lr_to_some_data)->attribute( 'CUSTOMER' )->to_equals(  '10101010' ).
```
It would help on structures too:
```
expect( ls_some_data )->attribute( 'CUSTOMER' )->to_equals( '101010' ). 
"instead of 
expect( ls_some_data)->to_equals( value some_structure( <all fields and values expected for this test scenario> ) ). 

```
Although, for structures I guess below is more than enough ;)
```
expect( ls_some_data-customer )->to_equals( '101010' ). 
```
Cheers! And best regards,
Felipe Silva

**Comment by eduardocopat** [comment]
Hi, Felipe,

Why couldn't you do `expect( lr_to_some_data->CUSTOMER )->to_equals(  '10101010' ).` ?

Considering of course that `CUSTOMER` is a public attribute.

**Comment by xinitrc86** [comment]
Hi Copat,

Yes, and in the end I refactored some parts so I could have typed references. 

I still use what I proposed on some parts (and tests) where I don't have a typed reference and deferencing it in the test looks weird. This is particularly true when working with BOPF and its service manager. And some code base around it...

I'm closing it as I'm not a fan of supporting the dynamic gimmicks that lead me to need it in the first place ;)

Thanks anyway.
Regards,
Felipe
